src/components/pages/dashboard/panels/kpis/kpisPanel.js

```diff
@@ -0,0 +1,216 @@
+      isPending: true,
+      currentActiveAlarms: [],
+      previousActiveAlarms: [],
+      currentAlarms: [],
+      previousAlarms: [],
+      criticalAlarmsChange: 0
+    };
+
+    // Initialize chart client
+    this.tsiClient = new window.TsiClient();
+  }
+
+  componentDidMount() {
+    // Create line chart
+    this.barChart = new this.tsiClient.ux.BarChart(document.getElementById(barChartId));
+    this.pieChart = new this.tsiClient.ux.PieChart(document.getElementById(pieChartId));
+
+    const currentFrom = 'NOW-PT1H';
+    const previousFrom = 'NOW-PT2H';
+
+    const currentParams = { from: currentFrom, to: 'NOW' };
+    const previousParams = { from: previousFrom, to: currentFrom };
+
+    // TODO: Add device ids to params - START
+    this.subscription = Observable.forkJoin(
+      TelemetryService.getActiveAlarms(currentParams), // Get current
+      TelemetryService.getActiveAlarms(previousParams), // Get previous
+
+      TelemetryService.getAlarms(currentParams),
+      TelemetryService.getAlarms(previousParams)
+    ).subscribe(([
+      currentActiveAlarms,
+      previousActiveAlarms,
+
+      currentAlarms,
+      previousAlarms
+    ]) => {
+      // ================== Critical Alarms - START
+      const currentCriticalAlarms = countCriticalAlarms(currentAlarms);
+      const previousCriticalAlarms = countCriticalAlarms(previousAlarms);
+      const criticalAlarmsChange = ((currentCriticalAlarms - previousCriticalAlarms) / currentCriticalAlarms * 100).toFixed(2);
+      // ================== Critical Alarms - END
+      this.setState({
+        currentActiveAlarms,
+        previousActiveAlarms,
+        currentAlarms,
+        previousAlarms,
+        criticalAlarmsChange,
+        isPending: false
+      });
+    });
+  }
+
+  componentWillUnmount() {
+    if (this.subscription) this.subscription.unsubscribe();
+  }
+
+  componentWillUpdate(nextProps, nextState) {
+    const time = '2017-04-19T13:00:00Z'; // TODO: Remove hard coded time, needed by the charts
+
+    // ================== Bar chart - START
+    if (nextState.currentActiveAlarms.length) {
+      // Compute top 5 kpis
+      const topAlarms = nextState.currentActiveAlarms
+        .sort(compareByProperty('count'))
+        .slice(0, maxTopAlarms);
+
+      // Find the previous counts for the current top 5 kpis
+      const previousTopAlarms = nextState.previousActiveAlarms.reduce(
+        (acc, { ruleId, count }) =>
+          (ruleId in acc)
+            ? { ...acc, [ruleId]: count }
+            : acc
+        ,
+        topAlarms.reduce((acc, { ruleId }) => ({ ...acc, [ruleId]: 0 }), {})
+      );
+
+      // Convert the raw counts into a chart readable format
+      const barChartDatum = topAlarms.map(({ ruleId, count }) => ({
+        [(this.props.rules[ruleId] || {}).name || ruleId]: {
+          'Current Month': { [time]: { val: count } }, // TODO: Translate legends
+          'Previous Month': { [time]: { val: previousTopAlarms[ruleId] } },
```

Comment:
Add to do locale.

Reply:
There is one on line 139

Reply:
ok wasn't sure what translate legends meant :)